qiskit/tools/parallel.py

```diff
@@ -105,9 +105,7 @@ def _task_wrapper(param):
     return task(value, *task_args, **task_kwargs)
 
 
-def parallel_map(  # pylint: disable=dangerous-default-value
-    task, values, task_args=tuple(), task_kwargs={}, num_processes=CPU_COUNT
-):
+def parallel_map(task, values, task_args=(), task_kwargs={}, num_processes=CPU_COUNT):  # noqa: B006
```

Comment:
Why disable the dangerous default warning?  It's valid here: it's a mistake to default `task_kwargs={}` because (hypothetical) mutations _shouldn't_ persist.  It's better to do `task_kwargs=None` and have `task_kwargs = task_kwargs if task_kwargs is not None else {}` (or similar) in the body.

Reply:
I assumed that the author intentionally wanted `task_kwargs` to persist. If the code remains as it is, there should be a test to ensure the unusual behavior. If it's really a mistake, I can remove it.

Reply:
It was originally done "correctly" https://github.com/Qiskit/qiskit-terra/commit/d7feb10aa11421641e0ebea40569ce7513bb3f74 , but was changed to the current form. There is no comment as to why. The commit message making the change mentions progress bars.

Reply:
@nonhermitian Do you recall why `task_kwargs` was first initialized to `None` and set to `{}` in the function body; and in a later commit was initialized to `{}`? Looks like you intentionally wanted a persistent `dict`.